src/gleam/erlang/file.gleam

```diff
@@ -109,8 +109,97 @@ pub type Reason {
   Exdev
   /// File was requested to be read as UTF-8, but is not UTF-8 encoded.
   NotUTF8
+  /// Path is not a directory
+  Badarg
 }
 
+/// Returns true if path refers to a directory, otherwise false.
+///
+/// ## Examples
+///
+///    > is_dir("/tmp")
+///    True
+///
+///    > is_dir("resume.pdf")
+///    False
+pub external fn is_dir(path: String) -> Bool =
+  "filelib" "is_dir"
+
+/// Returns true if path refers to a file, otherwise false.
+///
+/// ## Examples
+///
+///    > is_file("resume.pdf")
+///    True
+///
+///    > is_file("/tmp")
+///    True
+///
+///    > is_file("/does_not_exist")
+///    False
+pub external fn is_file(path: String) -> Bool =
+  "filelib" "is_file"
+
+/// Tries to create a directory. Missing parent directories are not created.
+///
+/// Returns a Result of nil if the directory is created or Reason if the
+/// operation failed.
+///
+/// ## Examples
+///
+///    > make_dir("/tmp/foo")
+///    Ok(Nil)
+///
+///    > make_dir("relative_dir")
+///    Ok(Nil)
+///
+///    > make_dir("/tmp/missing_intermediate_dir/foo")
+///    Error(Badarg)
+pub external fn make_dir(path: String) -> Result(Nil, Reason) =
+  "gleam_erlang_ffi" "make_dir"
+
+/// Lists all files in a directory, except files with raw filenames.
+///
+/// Returns a Result containing the list of filenames in the directory, or Reason
+/// if the operation failed.
+///
+/// ## Examples
+///
+///    > read("/tmp")
+///    Ok(["FB01293B-8597-4359-80D5-130140A0C0DE","AlTest2.out"])
+///
+///    > read("resume.doc")
+///    Error(Badarg)
+///
+///    > read("/missing/dir")
+///    Error(Badarg)
+pub external fn list_dir(path: String) -> Result(List(String), Reason) =
+  "gleam_erlang_ffi" "list_dir"
+
+/// Deletes a directory.
+///
+/// The directory must be empty before it can be deleted. Returns a nil Success
+/// or Reason if the operation failed.
+///
+/// ## Examples
+///
+///    > del_dir("foo")
+///    Ok(Nil)
+///
+pub external fn delete_dir(path: String) -> Result(Nil, Reason) =
+  "gleam_erlang_ffi" "delete_dir"
+
+/// Deletes a file or directory recursively.
+///
+/// Returns a nil Success or Reason if the operation failed.
+///
+/// ## Examples
+///
+///    > delete_dir("foo")
+///    Ok(Nil)
+pub external fn delete_dir_r(path: String) -> Result(Nil, Reason) =
```

Comment:
As it can delete files perhaps `recursive_delete` is a more fitting name

Reply:
Fixed